lib/model-builder.js

```diff
@@ -308,7 +308,9 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
   // Add the id property
   if (idInjection) {
     // Set up the id property
-    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true});
+    // updateOnly property is added to indicate that this property will appear in the model for update/updateorcreate
+    // operations but and not for create operation.
+    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true, updateOnly: true});
```

Comment:
Agree ^^

I tried
```
  var forceId = ModelClass.settings.forceId;

  if (idNames.length > 0) {
    // id already exists
    idInjection = false;
    // Does method `idName` also works for composite Id at this phase?
    var idName = modelDefinition.idName();
    var idProp = ModelClass.definition.rawProperties[idName];
    if (idProp.generated && forceId !== false) {
      forceId = true;
    }
  } else {
    if (forceId !== false) forceId = true;
  }
  ModelClass.settings.forceId = forceId;
```
received same errors as @rashmihunt is fixing in this PR, other tests pass.